editor/src/components/titlebar/Buttons.tsx

```diff
@@ -0,0 +1,64 @@
+import styled from '@emotion/styled'
+import React from 'react'
+import { Button } from '../../uuiui'
+
+interface TitleButtonProps {
+  onClick: () => void
+  color?: string
+}
+
+const TitleButton = styled(Button)((props) => ({ fontWeight: 500, fontSize: 10 }))
+
+export const FullHeightButton: React.FC<React.PropsWithChildren<TitleButtonProps>> = ({
+  onClick,
+  children,
+}) => {
+  const ButtonEl = styled(TitleButton)((props) => ({
+    padding: '1em 1.5em',
+    height: '100%',
+    borderRadius: 0,
+  }))
+
+  return (
+    <ButtonEl primary={true} highlight={true} onClick={onClick}>
+      {children}
+    </ButtonEl>
+  )
+}
+
+export const TextButton: React.FC<React.PropsWithChildren<TitleButtonProps>> = ({
+  onClick,
+  children,
+}) => {
+  const ButtonEl = styled(TitleButton)((props) => ({
+    padding: '1em 1.5em',
+    height: '100%',
+    borderRadius: 20,
+    fontSize: 12,
+  }))
+
+  return (
+    <ButtonEl highlight={true} onClick={onClick}>
+      {children}
+    </ButtonEl>
+  )
+}
+
+export const RoundedButton: React.FC<React.PropsWithChildren<TitleButtonProps>> = ({
+  onClick,
+  color,
+  children,
+}) => {
+  const ButtonEl = styled(TitleButton)((props) => ({
+    padding: '3px 12px 3px 8px',
+    height: 36,
+    borderRadius: 18,
+    ...(color && { backgroundColor: color }),
```

Comment:
Wait, what's going on here? Is this spreading a boolean?

Reply:
It's valid syntax! If color is set, you get the spread object, but if it it's false, you get nothing. But yes, you can actually spread a boolean.

Reply:
The `&&` operator will return the right value if the left value is truthy, so this is checking if `color` is defined, and then returning the object on the right of that `&&`, and then spreading that. Please can we avoid this syntax, as it can be pretty confusing (and we are very much against truthiness as it has bit us on many occasions), and instead be more explicit:
```ts
const bgColor = color == null ? {} : { backgroundColor: color }
const ButtonEl = styled(TitleButton)((props) => ({
    padding: '3px 12px 3px 8px',
    height: 36,
    borderRadius: 18,
    ...bgColor,
}))
```